RELEASE.md

```diff
@@ -9,6 +9,39 @@ This file is copied to nebari-dev/nebari-docs using a GitHub Action. -->
 
 ---
 
+### Release 2024.7.1 - July 29, 2024
```

Comment:
@viniciusdc do we want to use today's date or leave this one?

Reply:
good catch